plugins/lsp/language-servers/default.nix

```diff
@@ -389,10 +389,64 @@ let
             '';
           };
         };
+        nix = {
+          binary = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = ''
+              The path to the `nix` binary.
+            '';
+          };
+          maxMemoryMB = mkOption {
+            type = types.nullOr types.int;
+            default = null;
+            description = ''
+              The heap memory limit in MiB for `nix` evaluation.
+              Currently it only applies to flake evaluation when `autoEvalInputs` is
+              enabled, and only works for Linux. Other `nix` invocations may be also
+              applied in the future. `null` means no limit.
+              As a reference, `nix flake show --legacy nixpkgs` usually requires
+              about 2GiB memory.
+            '';
+          };
+          flake = {
+            autoArchive = mkOption {
+              type = types.nullOr types.bool;
+              default = null;
+              description = ''
+                Auto-archiving behavior which may use network.
+                 - null: Ask every time.
+                 - true: Automatically run `nix flake archive` when necessary.
+                 - false: Do not archive. Only load inputs that are already on disk.
+              '';
+            };
+            autoEvalInputs = mkOption {
+              type = types.nullOr types.bool;
+              default = null;
+              description = ''
+                Whether to auto-eval flake inputs.
+                The evaluation result is used to improve completion, but may cost
+                lots of time and/or memory.
+              '';
+            };
+            nixpkgsInputName = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              description = ''
+                The input name of nixpkgs for NixOS options evaluation.
+
+                The options hierarchy is used to improve completion, but may cost
+                lots of time and/or memory.
+                If this value is `null` or is not found in the workspace flake's
+                inputs, NixOS options are not evaluated.
+              '';
+            };
+          };
+        };
       };
       settings = cfg: {
         nil = {
-          inherit (cfg) formatting diagnostics;
+          inherit (cfg) formatting diagnostics nix;
         };
```

Comment:
> options that were being ignored when I tried to add to config for `nil_ls`

Should this just be `nil = cfg;`, is inheriting specific options the reason why this didn't work as-per RFC42?

@GaetanLepage @traxys

Reply:
Makes sense, I tested that built my config still and supports new attribute sets not defined.